File: TFC/TFC.Application.DTO/EntityDTO/UserDTO.cs
namespace TFC.Application.DTO.EntityDTO
{
    public class UserDTO
    {
        public long UserId { get; set; }
        public string? Dni { get; set; }
        public string? Username { get; set; }
        public string? Password { get; set; }
        public string? Surname { get; set; }
        public string? Email { get; set; }
        public string? Role { get; set; } 
        public DateTime InscriptionDate { get; set; }
        public List<RoutineDTO> Routines { get; set; } = new List<RoutineDTO>();
    }
}

File: tfc_gym_app/lib/data/models/dto/entitie/user_dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:tfc_gym_app/data/models/enums/role.dart';

part 'user_dto.g.dart';

@JsonSerializable()
class UserDTO {
  final int userId;
  final String? dni;
  final String? username;
  final String? surname;
  final String? email;
  final String? friendCode;
  final String password;
  final int routinesCount;
  final int friendsCount;
  final DateTime? inscriptionDate; 
  final Role? role;

  const UserDTO({
    required this.userId,
    this.dni,
    this.username,
    this.surname,
    this.email,
    this.friendCode,
    this.password = "********",
    this.routinesCount = 0,
    this.friendsCount = 0,
    this.inscriptionDate, 
    this.role,
  });

  factory UserDTO.fromJson(Map<String, dynamic> json) => _$UserDTOFromJson(json);
  Map<String, dynamic> toJson() => _$UserDTOToJson(this);
}